phpBB/phpbb/attachment/upload.php

```diff
@@ -315,6 +317,16 @@ protected function check_disk_space()
 				return false;
 			}
 		}
+		else
+		{
+			$this->file_data['error'][] = $this->language->lang('ATTACH_DISK_FREE_SPACE');
```

Comment:
If it's not disabled we only give out system specific info for admins. Therefore I don't see the point of telling anyone that this function is disabled.

Reply:
Previous behavior was to silently fail if we can't check the free space. This can be accepted behavior and changing this to prevent all uploads is IMHO not correct.

Reply:
And let the storage folder increase in size since the TMP files are not deleted?

Reply:
> If it's not disabled we only give out system specific info for admins. Therefore I don't see the point of telling anyone that this function is disabled.

At least provides a meaningful error message, what would it be your purpose?

Reply:
This error message can only be meaningful to admins and would just confuse users. Also, the previous behavior was different and I don't see a need to change it. You're aborting upload because you don't know if there is free space as opposed to aborting because there is not enough space. This change will essentially break uploads with boards that do not support calling disk_free_space().

Reply:
Options:

1. add an extra language string and present the error message different for admins/users via $auth->acl_get()
2. add a new configuration in ACP to check if the disk space should be checked or not (NO as defaut)
3. remove the else statement (which I do not like)

Reply:
I think the big question is how one would end up at this else branch. In this case it's only if the function does not exist, hence a state when you are not able to check the disk space. I think the decision to stop any and all uploads because I'm not able to check this is wrong. Yes, you might show an admin that this is disabled. Then again, if this is disabled it might be an issue but most of the time it would be perfectly fine. Breaking uploads because we're not able to check something just seems wrong to me.

Reply:
This has in fact been the behaviour of the previous code so far as it has been silenced. Nowadays I don't think there are any space limitations and so I could agree with you, but since there are still some sites that offer free web space limited to e.g. 5GB and which many people use I think there could be problems in the future. Let's decide together which is the best solution and I'll be happy to push any changes.

Reply:
I think in general it would probably be nice to have a system where admins gets informed about something like that, e.g. via notifications. But in general I think it does not make sense to abort the actual uploading of attachments and therefore the posting for this. Something like a notification "hey, your system is running low on free space" but that will exceed the scope of this issue / PR.